Switch multiblocks to MBTweaker

eutro opened the pull request:
Quests aren't updated yet.

This *includes* the minimum CT update required for functional functional interfaces, though it should be updated to a stable release when one comes out.

eutro committed:
Switch multiblocks to MBTweaker

Quests aren't updated yet.

This *includes* the minimum CT update required for functional functional interfaces, though it should be updated to a stable release when one comes out.

ALongStringOfNumbers commented:
Note that the Crafttweaker update is also included in #476

kiljacken commented on overrides/scripts/Microverse.zs:
```diff
@@ -18,164 +18,50 @@ import scripts.CommonVars.makeShapeless3 as makeShapeless3;
+//                 XXXXXXXXXX                            XXXXXXXXXX
+//              XXXXXXXXXXXXXXX                        XXXXXXXXXXXXXXX
+//            XXXXXXXXXXXXX XXXX                      XXXX XXXXXXXXXXXXX
+//           XXXXXXXXXXXXX XXXXX                      XXXXX XXXXXXXXXXXXX
+//          XXXXXXXXXXXXX XXXXXX                      XXXXXX XXXXXXXXXXXXX
+//         XXXXXXXXXXXXXXXXXXXXX                      XXXXXXXXXXXXXXXXXXXXX
+//        XXXXXXXXXXXXXXXXXXXXXX                      XXXXXXXXXXXXXXXXXXXXXX
+//       XXXXXXXXXXXXXXXXXXXXXX                        XXXXXXXXXXXXXXXXXXXXXX
+//       XXXXXXXXXXXXXXXXXXXXX  XXXXX            XXXXX  XXXXXXXXXXXXXXXXXXXXX
+//      XXXXXXXXXXXXXXXXXXXXX  XXXXXXXX        XXXXXXXX  XXXXXXXXXXXXXXXXXXXXX
+//      XXXXXXXXXXXXXXXXXXX    XXXXXXXX        XXXXXXXX    XXXXXXXXXXXXXXXXXXX
+//      XXXXXXXXXXXXXXXXX      XXXXXXXXX      XXXXXXXXX      XXXXXXXXXXXXXXXXX
+//      XXXXXXXXXXXXXXX          XXXXXXXXXXXXXXXXXXXX          XXXXXXXXXXXXXXX
+//      XXXXXXXXXXXX          XXXXXXXXXXXXXXXXXXXXXXXXXX          XXXXXXXXXXXX
+//      XXXXXXXXXX        XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX        XXXXXXXXXX
+//       XXXXXXXXXX     XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX     XXXXXXXXXX
+//        XXXXXXXXXX  XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX  XXXXXXXXXX
+// XXXX     XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX     XXXX
+// XXXXXX    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX    XXXXXX
+// XXXXXXXX    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX    XXXXXXXX
+//  XXXXXXXX     XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX    XXXXXXXX
+//   XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+//    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+//       XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+//          XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+// XXXXXXXXX  XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX  XXXXXXXXX
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+// XXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+//    XXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXX
+//         XXX    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX    XXX
+//            XXXXXXXXXXXX  XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX  XXXXXXXXXXX
+//       XXXXXXXXXXXXXXXX  XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX  XXXXXXXXXXXXXXXX
+//    XXXXXXXXXXXXXXXXX   XXXXXXXX     XXXXXXXX     XXXXXXXX   XXXXXXXXXXXXXXXXX
+//    XXXXXXXXXXXXXX    XXXXXXXXX                    XXXXXXXXX    XXXXXXXXXXXXXX
+//     XXXXXXXXXX      XXXXXXXX                       XXXXXXXXX      XXXXXXXXXX
+//                     XXXXXXXX                        XXXXXXXX
+//                     XXXXXXXX                        XXXXXXXX
+//                      XXXXXXX                        XXXXXXX
+//                      XXXXXX                          XXXXXX
+//                      XXXXX                            XXXXX
```
What's up with all the crabs?

eutropius225 commented:
It's to mark things that need to go when Modular Machinery gets removed.

eutro committed:
Conversion recipes between controllers.

NotMyWing committed:
Update "Small Microverse Projector"

NotMyWing committed:
Update "Cryogenic Air Distillation"

NotMyWing committed:
Update "Medium Microverse Projector"

NotMyWing committed:
Update "Large Microverse Projector"

NotMyWing committed:
Update "Naquadah Reactor Mk1/Mk2"

NotMyWing committed:
Update "Lunar Mining Station"

NotMyWing committed:
Update "Oil Drilling Rig"

NotMyWing committed:
Revert "Oil Rig Rebalance" (#348)

Microverse multiblocks can now be overclocked at the cost of power efficiency. Fixes #433.

NotMyWing committed:
Adjust Quest Dependencies

* "Small Microverse Projector" now also depends on "The First Tier Four Circuit"
* "Medium Microverse Projector" now also depends on "The First Tier Four Circuit"
* "Large Microverse Projector" now also depends on "The First Tier Five Circuit"
* "Oil Drilling Rig" now depends on "The First Tier Three Circuit" and "Stainless Steel"

NotMyWing committed:
Create "Multiblock Machine Previews"

A new quest to give a quick overview of the in-world preview mechanic.
Add hints to several multiblock quests that might be confusing to build.

eutro committed:
Merge remote-tracking branch 'origin/MBTweaker' into MBTweaker

NotMyWing commented:
@Exaxxion proofreading pls

NotMyWing committed:
Fix Indents

what's up with these mile-long indents???????????

eutro committed:
Update Multiblock Tweaker

eutro committed:
Merge remote-tracking branch 'origin/MBTweaker' into MBTweaker

smudgerox commented:
double negative on line 14595 of DefaultQuests.json

which includes but isn\u0027t not limited to

should be 

which includes but is not limited to

ALongStringOfNumbers reviewed (commented):
To prevent confusion, especially from newer players to the pack, I believe the controller conversion recipe should be just from the MM controller, not from the oredict of all the controllers. In addition, each of the new controllers should have their own independent recipe.

ALongStringOfNumbers commented on overrides/scripts/Multiblocks.zs:
```diff
@@ -0,0 +1,1643 @@
+    .inputs(<inspirations:pipe>)
+    .fluidInputs(<liquid:drilling_fluid> * 10)
+    .fluidOutputs(<liquid:oil> * 1000)
+	.property("consumeChance", 1)
```
The original Consumption chance for the Pipe was 4%

ALongStringOfNumbers replied:
This was increased during the buff, which we are reverting, so never mind this.

ALongStringOfNumbers committed:
Slight tweak of comment

ALongStringOfNumbers committed:
Another slight comment tweak

ALongStringOfNumbers committed:
Address review comment, fixing double negative in description

ALongStringOfNumbers commented on overrides/scripts/Multiblocks.zs:
```diff
@@ -0,0 +1,1643 @@
+    .withRecipeMap(
+        FactoryRecipeMap.start(loc)
+                        .minInputs(2)
+                        .maxInputs(6)
```
Mentioned by Exa, the maximum number of inputs for any MMP recipe is 4.

Commit:
Show 4 input slots in the MMP JEI category.

Exaxxion committed:
Quest Adjustments
Change some of the quest text, remove energy hatches from some of the quests
to let players choose which they want to use, now that you can't footgun with
too small of a MM energy hatch.

eutro committed:
Update MBTweaker to beta 0.1.3, add max and min requirements.
Use the fancy IBlockMatcher.IN_PREVIEW.

eutro committed:
Merge remote-tracking branch 'origin/MBTweaker' into MBTweaker

eutro committed:
Add cute rocket arrow, update mod, fix some formatting.

NotMyWing commented on overrides/config/betterquesting/DefaultQuests.json:
```diff
@@ -26646,7 +26646,7 @@
         "betterquesting:10": {
           "autoclaim:1": 0,
-          "desc:8": "The §3Medium Microverse Projector§r is a more powerful projector required for sending Tier Four through Tier Six Micro Miners into microverses.\n\nWhile some of the miners run on IV power, make a §6LuV Energy Input§r so you don\u0027t have to replace it later.\n\n§6Sneak-right click§r the controller to enable the in-world preview.",
+          "desc:8": "The §3Medium Microverse Projector§r is a more powerful projector required for sending §6Tier Four§r through §6Tier Six Micro Miners§r into microverses.\n\nInitial missions can operate on §eIV power§r, while §eLuV power§r is required for later missions. As with the prior projector, this structure supports overclocking and the position of buses and hatches can be swapped with a ...
```
`§6Tier Four§r through §6Tier Six Micro Miners§r` has color tags, but

NotMyWing commented on overrides/config/betterquesting/DefaultQuests.json:
```diff
@@ -26949,7 +26943,7 @@
         "betterquesting:10": {
           "autoclaim:1": 0,
-          "desc:8": "The §3Large Microverse Projector§r is a more powerful projector required for sending Tier Seven through Tier Ten Micro Miners into microverses.\n\nWhile some of the miners run on lower power, make a §6UV Energy Input§r so you don\u0027t have to replace it later or accidentally run something on insufficient power.\n\n§6Sneak-right click§r the controller to enable the in-world  ...
+          "desc:8": "The §3Large Microverse Projector§r is a more powerful projector required for sending Tier Seven through Tier Ten Micro Miners into microverses.\n\nTier Seven missions will operate on §eLuV power§r, but higher energy tiers will be required for later missions. As with the prior projector, this structure supports overclocking and the position of buses and hatches can be swapped  ...
```
`Tier Seven through Tier Ten Micro Miners` doesn't

Exaxxion commented:
Good catch. We should add them to the second set.

Exaxxion committed:
Fix quest text highlighting

Exaxxion committed:
Rewrite JEI preview descriptions for custom multiblocks

Exaxxion reviewed (approved):
At this time I think everything looks good. We will wait for @eutropius225 to do a stable release of the mod, update to it, then proceed with merging. Ideally CT will release the functionality we're using in the beta version as a full release too.

NotMyWing commented on overrides/resources/pack.mcmeta:
```diff
@@ -1 +1 @@
-{"pack":{"pack_format":3,"description":"Omnifactory Resources"}}
\ No newline at end of file
+{"pack":{"pack_format":3,"description":"B.A.S.E External Resources"}}
```
?

eutro replied:
That's BASE being stupid, I suppose it ought to be done away with.

eutro committed:
Revert BASE being dumb

eutro committed:
Update to stable release.

eutro committed:
Merge remote-tracking branch 'origin/MBTweaker' into MBTweaker

eutro committed:
Update to latest MBTweaker

Exaxxion committed:
Update CT to latest beta

Exaxxion commented:
Before merging this, I'd like to investigate implementing Microversium as a custom GTCE material instead of ContentTweaker items. I think it could give us more flexibility to expand on or adjust the recipes later, and doing so now would avoid a second release-staggered item conversion like we're already having to do for Modularium.

eutro committed:
Update to latest.

eutro committed:
Merge remote-tracking branch 'origin/MBTweaker' into MBTweaker

# Conflicts:
#	manifest.json

eutro committed:
Update again.

Exaxxion committed:
Switch Microversium to GTCE Material
- Use GTCE Material instead of adding via ContentTweaker. GTCE auto-generates
  components, dusts, block, and fluid forms for us.
- Switch recipes to use GTCE version.
- Add recipe for long rod in the lathe.
- Minor refactoring.

Exaxxion reviewed (approved):
I think this is good to go. Would appreciate a look-over from everyone to see if we overlooked anything or made a mistake somewhere.

ALongStringOfNumbers reviewed (commented):
The Quest for the Small Microverse Projector does not have the line about sneak + right click to show the in world preview that was added to other custom multiblocks. This line is not present in the quests for the regular gregtech multiblocks either, although the functionality was added. Other than that, looks good except for maybe removing some crabs.

Edit: I just saw the new added quest mentioning the previews, but maybe still a reminder line for the Small projector. In addition, would it be worth it to mention the enhanced zoom and pan capabilities for the custom multiblocks in this quest? (Right click pan, shift + right click zoom)

ALongStringOfNumbers committed:
Adjust Quest wording

Exaxxion commented:
Now that we all agree the new stuff looks good, let's remove the MM structures and recipes so they don't show up in JEI (as it would be confusing) and to ensure players make the switch before the subsequent release.

We should also implement conversion recipes from the MM power hatches to the GTCE one used to craft it (shapeless table recipe should be fine) since we seem to have forgotten to do that.

Exaxxion committed:
Remove machine definitions for ModularMachinery multiblocks

Exaxxion committed:
Switch Modularium quest in Endgame Tab to Microversium
Apparently also some whitespace changes done by bq_admin default save.

Exaxxion committed:
Recipe cleanup and more conversions
* Remove recipes defined for all ModularMachinery multiblocks
* Add recipes to recover the most important ingredient from each kind of bus
  and hatch, to make the switchover easier.

Exaxxion commented:
I added a few more conversion/refund recipes for the various MM hatches and buses, fixed an issue with the Endgame tab quest still wanting Modularium, and removed the structure definitions and recipes for MM blocks. With that I think the conversion is complete.

Feel free to give it another lookover, and barring any issues I'll squash and merge this thing.

ALongStringOfNumbers commented on overrides/config/betterquesting/DefaultQuests.json:
```diff
@@ -47119,7 +47119,7 @@
         "betterquesting:10": {
           "autoclaim:1": 0,
-          "desc:8": "§bMultiblock Tweaker§r adds an §einteractive in-world preview§r for all §bGregTech§r multiblocks! §6Shift-right-click§r a placed controller to activate a hologram depicting the default layout of the structure to help you with building. As you place blocks, the structure will be validated and the §efirst incorrect block§r it detects will be highlighted in red.\n\n§6Sneak-right ...
+          "desc:8": "You can §6sneak-right click§r the controllers of §bGregTech§r multiblock machines with an empty hand to enable §ethe interactive in-world preview§r! This will show you a hologram of how a machine must be built. However, it has a small quirk of displaying §ethe first errored block only§r.\n\n§6Sneak-right click§r again to toggle the per-layer display.\n\n§6Sneak-left click§r t ...
```
Changes here

Exaxxion replied:
Yeah I goofed. I need to revert the changes to quest text. I thought it was whitespace but I didn't look closely enough.

Exaxxion committed:
Revert inadvertent changes to quest text in commit e2ce76ef

Exaxxion commented:
Ok, the quest text should be fixed now. Please let me know if you see anything else.
